metrique/tests/ui/fail/bad_metrics_value.stderr

```diff
@@ -107,6 +107,19 @@ help: consider importing one of these structs
 1  + use metrique_writer_core::unit::Second;
    |
 
+error[E0277]: the trait bound `String: SampleGroup` is not satisfied
```

Comment:
should we add something like:
```
#[diagnostic::on_unimplemented(
    message = "`{Self}` cannot be used as a sample group",
    note = "sample groups must implement `SampleGroup`",
    note = "consider using `&'static str` instead of `String`, or implement `SampleGroup` for `{Self}`"
)]
```

to improve this error?

Reply:
We can, I didn't think it was important enough, wil ladd